src/tribler/core/utilities/sqlite_utils.py

```diff
@@ -0,0 +1,241 @@
+from __future__ import annotations
+
+import logging
+import sqlite3
+import sys
+from contextlib import contextmanager
+from pathlib import Path
+from sqlite3 import DataError, DatabaseError, Error, IntegrityError, InterfaceError, InternalError, NotSupportedError, \
+    OperationalError, ProgrammingError, Warning, sqlite_version_info  # pylint: disable=unused-import, redefined-builtin
+from typing import Any, Generator, List, Literal, TypeVar, Union
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    Self = TypeVar("Self")
+
+
+# This module serves as a replacement to the sqlite3 module and handles the case when the database is corrupted.
+# It provides the `connect` function that should be used instead of `sqlite3.connect` and the `Cursor` and `Connection`
+# classes that replaces `sqlite3.Cursor` and `sqlite3.Connection` classes respectively. If the `connect` function or
+# any Connectoin or Cursor method is called and the database is corrupted, the database is marked as corrupted and
+# the DatabaseIsCorrupted exception is raised. It should be handled by terminating the Tribler Core with the exit code
+# EXITCODE_DATABASE_IS_CORRUPTED (99). After the Core restarts, the `handle_db_if_corrupted` function checks the
+# presense of the database corruption marker and handles it by removing the database file and the corruption marker.
+# After that, the database is recreated upon the next attempt to connect to it.
+
+
+logger = logging.getLogger(__name__)
+
+
+class DatabaseIsCorrupted(Exception):
+    pass
+
+
+@contextmanager
+def _handling_malformed_db_error(db_filepath: Path):
```

Comment:
Private methods should be placed after public ones.

Reply:
Done